genrl/deep/agents/sac/sac.py

```diff
@@ -267,112 +276,105 @@ def select_action(self, state):
+                    action.reshape(-1, *self.env.action_space.shape).float(),
+                    next_state.reshape(-1, *self.env.observation_space.shape).float(),
+                )
+                reward, done = reward.reshape(-1, 1), done.reshape(-1, 1)
 
-        q1_loss = nn.MSELoss()(q1, next_q)
-        q2_loss = nn.MSELoss()(q2, next_q)
+            with torch.no_grad():
+                next_action, next_log_pi, _ = self.sample_action(next_state)
+                next_q1_targ = self.q1_targ(
+                    torch.cat([next_state, next_action], dim=-1)
+                )
+                next_q2_targ = self.q2_targ(
+                    torch.cat([next_state, next_action], dim=-1)
+                )
+                next_q_targ = (
+                    torch.min(next_q1_targ, next_q2_targ) - self.alpha * next_log_pi
+                )
+                next_q = reward + self.gamma * (1 - done) * next_q_targ
 
-        pi, log_pi, _ = self.sample_action(state)
-        q1_pi = self.q1(torch.cat([state, pi.float()], dim=-1).float())
-        q2_pi = self.q2(torch.cat([state, pi.float()], dim=-1).float())
-        min_q_pi = torch.min(q1_pi, q2_pi)
-        policy_loss = ((self.alpha * log_pi) - min_q_pi).mean()
+            # compute losses
+            q1 = self.q1(torch.cat([state, action], dim=-1))
+            q2 = self.q2(torch.cat([state, action], dim=-1))
 
-        # gradient step
-        self.q1_optimizer.zero_grad()
-        q1_loss.backward()
-        self.q1_optimizer.step()
+            q1_loss = nn.MSELoss()(q1, next_q)
+            q2_loss = nn.MSELoss()(q2, next_q)
 
-        self.q2_optimizer.zero_grad()
-        q2_loss.backward()
-        self.q2_optimizer.step()
+            pi, log_pi, _ = self.sample_action(state)
+            q1_pi = self.q1(torch.cat([state, pi.float()], dim=-1).float())
+            q2_pi = self.q2(torch.cat([state, pi.float()], dim=-1).float())
+            min_q_pi = torch.min(q1_pi, q2_pi)
+            policy_loss = ((self.alpha * log_pi) - min_q_pi).mean()
 
-        self.policy_optimizer.zero_grad()
-        policy_loss.backward()
-        self.policy_optimizer.step()
+            # gradient step
+            self.q1_optimizer.zero_grad()
+            q1_loss.backward()
+            self.q1_optimizer.step()
 
-        # alpha loss
-        if self.entropy_tuning:
-            alpha_loss = -(
-                self.log_alpha * (log_pi + self.target_entropy).detach()
-            ).mean()
+            self.q2_optimizer.zero_grad()
+            q2_loss.backward()
+            self.q2_optimizer.step()
 
-            self.alpha_optim.zero_grad()
-            alpha_loss.backward()
-            self.alpha_optim.step()
+            self.policy_optimizer.zero_grad()
+            policy_loss.backward()
+            self.policy_optimizer.step()
 
-            self.alpha = self.log_alpha.exp()
-        else:
+            # alpha loss
             alpha_loss = torch.tensor(0.0).to(self.device)
```

Comment:
If you delete this line, it should get away with the LGTM error, right?

Reply:
yea it would go away i get that, but if you see the whole function we are returning alpha_loss so thought shouldn't remove that and pass it to get_logging_params instead

Reply:
Ok cool, I get it now. I'll get rid of it in my PR